### ISSUE TITLE
installation problem on MacOS due to fixed PyYaml version

### ISSUE THREAD

**Issue by weakcamel** [egnyte/gitlabform#30]
When installing gitlabform on MacOS (High Sierra), I'm facing problems.

As it happens, the specified version (3.12) is affected by problems described in https://github.com/yaml/pyyaml/issues/15

therefore *gitlabform* fails to install with pip as well.

The solution seems to be simple - update the dependency to pyyaml 3.13 which solves the problem, or don't requre a specific version in the first place.

**Comment by weakcamel** [comment]
Hello,
I'm happy to provide a PR if need be; just checking though - I'm not sure how active this project is; is there a chance that it would be accepted / released?

**Comment by gdubicki** [comment]
`gitlabform` does install on my MacOS High Sierra (10.13.6 (17G65)) without issues:

```
(test) gdubicki@mac ~ $ pip install gitlabform
Collecting gitlabform
  Using cached https://files.pythonhosted.org/packages/cf/bb/602df59fdd89213901196ad9220eb2fc28830ff5abed520780bf9da7cf06/gitlabform-1.0.0-py3-none-any.whl
Collecting pyyaml==3.12 (from gitlabform)
Collecting requests==2.18.4 (from gitlabform)
  Using cached https://files.pythonhosted.org/packages/49/df/50aa1999ab9bde74656c2919d9c0c085fd2b3775fd3eca826012bef76d8c/requests-2.18.4-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests==2.18.4->gitlabform)
  Downloading https://files.pythonhosted.org/packages/df/f7/04fee6ac349e915b82171f8e23cee63644d83663b34c539f7a09aed18f9e/certifi-2018.8.24-py2.py3-none-any.whl (147kB)
    100% |████████████████████████████████| 153kB 2.8MB/s
Collecting urllib3<1.23,>=1.21.1 (from requests==2.18.4->gitlabform)
  Using cached https://files.pythonhosted.org/packages/63/cb/6965947c13a94236f6d4b8223e21beb4d576dc72e8130bd7880f600839b8/urllib3-1.22-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests==2.18.4->gitlabform)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.7,>=2.5 (from requests==2.18.4->gitlabform)
  Using cached https://files.pythonhosted.org/packages/27/cc/6dd9a3869f15c2edfab863b992838277279ce92663d334df9ecf5106f5c6/idna-2.6-py2.py3-none-any.whl
Installing collected packages: pyyaml, certifi, urllib3, chardet, idna, requests, gitlabform
Successfully installed certifi-2018.8.24 chardet-3.0.4 gitlabform-1.0.0 idna-2.6 pyyaml-3.12 requests-2.18.4 urllib3-1.22
```

```
(test) gdubicki@mac ~ $ gitlabform -h
usage: gitlabform [-h] [-c CONFIG] [-v | -d] [--strict]
                  [--start-from START_FROM] [-n]
                  project_or_group

Easy configuration as code tool for GitLab using config in plain YAML.

(...)
```

..but if upgrade of pyyaml will help you and won't break anything then that PR is still very welcome.

**Comment by weakcamel** [comment]
Oops, sorry - I lost this one out of my sight for a while.
Many thanks for the fix and I can confirm it's solving the problem!

A side quesion: while pinning the dependencies on specific versions improves stability of `gitlabform`, it also freezes them for the user forever.

It's not a big deal of course, one can alsways use `virtualenv` but it might be a bit of an issue for OS-installed  or global pip packages.

What do you think about loosening it a bit and requiring *minimum* versions of the dependencies?

**Comment by gdubicki** [comment]
I could do that but this could cause me dealing with reports of bugs related to various, incompatible sets of dependencies installed on users machines.

I think that a better solution would be to provide a package with gitlabform for a few Linux distros, Mac and maybe Windows.

I'll create myself a ticket for that, but as usual help would be most welcome. Perhaps you could create a package for homebrew, @weakcamel ?

**Comment by weakcamel** [comment]
That's fair enough. 

I'm sure a package for Linux would be welcome by many.

If you mean it as alternative for relaxing dependencies, I'm not sure I follow. If the same dependencies to specific versions are repeated in e.g. an RPM, wouldn't it lock users to those versions even further?

**Comment by gdubicki** [comment]
I am thinking about RPM that would use virtualenv to not use system wide Python packages.

**Comment by weakcamel** [comment]
Yes, that would work however at a cost of size and making packages per arch/platform, unless you mean to skip the binaries somehow.

Dunno :) Might help some people, personally I'm happy with the lightweight pip version.

**Comment by gsemet** [comment]
Hi. Always install in a virtualenv. You can use pipenv to manage it automatically. Work great
